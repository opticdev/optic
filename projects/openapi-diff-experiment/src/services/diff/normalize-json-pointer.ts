import { jsonPointerHelpers } from '@useoptic/json-pointer-helpers';

export function normalizeJsonPointer(jsonPointer: string): string {
  const normalizedComponents = jsonPointerHelpers
    .decode(jsonPointer)
    .map((component) => {
      const isArrayIndex = !isNaN(component as any);
      if (isArrayIndex) {
        return '0';
      } else {
        return component;
      }
    });

  return jsonPointerHelpers.compile(normalizedComponents);
}
