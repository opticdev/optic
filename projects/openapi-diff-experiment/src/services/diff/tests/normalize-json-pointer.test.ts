import { normalizeJsonPointer } from '../normalize-json-pointer';

describe('normalize json pointer', () => {
  it('works for single array item', () => {
    expect(normalizeJsonPointer('/items/12')).toMatchSnapshot();
  });
  it('works for nested array items', () => {
    expect(
      normalizeJsonPointer('/items/1/properties/12/values/0')
    ).toMatchSnapshot();
  });
});
