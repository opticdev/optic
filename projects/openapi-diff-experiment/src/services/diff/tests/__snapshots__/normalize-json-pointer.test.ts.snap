// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize json pointer works for nested array items 1`] = `"/items/0/properties/0/values/0"`;

exports[`normalize json pointer works for single array item 1`] = `"/items/0"`;
